File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int goal = size(nums)-1;

        for(int i = size(nums)-1; i >= 0; i--){
            int length = i + nums[i];

            if(length >= goal){
                goal = i;
            }
        }
        return goal == 0;
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        goal = len(nums)-1

        for i in range(len(nums)-1,-1,-1):
            fastest = i + nums[i]
            if fastest >= goal:
                goal = i
        
        return goal == 0


        
        